### ISSUE TITLE
[BUG] quantile API mismatch with pandas

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#2045]
**Describe the bug**
We seem to have a parameter mismatch for `quantile` method of cudf & pandas.

pandas:
`DataFrame.quantile(q=0.5, axis=0, numeric_only=True, interpolation='linear')`

Cudf:
`DataFrame.quantile(self, q=0.5, interpolation='linear', columns=None, exact=True)`

**Steps/Code to reproduce bug**

This set of code works with pandas & dask

```python
df = pd.DataFrame({'C': [1, 2, 3]})
pddf = dask.dataframe.from_pandas(df, npartitions=1)
pddf.describe().compute()
```
output:
```python
    C    0
count    3.0    NaN
mean    2.0    NaN
std    1.0    NaN
min    1.0    NaN
25%    NaN    1.5
50%    NaN    2.0
75%    NaN    2.5
max    3.0    NaN
```

But fails in dask cudf due to parameter mismatch:

```python
df = cudf.DataFrame({'C': [1, 2, 3]})
ddf = dask_cudf.from_cudf(df,npartitions=1)
ddf.describe().compute()
```
exception:
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-46-f706cb334076> in <module>
      1 df = cudf.DataFrame({'C': [1, 2, 3]})
      2 ddf = dask_cudf.from_cudf(df,npartitions=1)
----> 3 ddf.describe().compute()

~/anaconda3/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in describe(self, split_every, percentiles, percentiles_method)
   1557                  num.std(split_every=split_every),
   1558                  num.min(split_every=split_every),
-> 1559                  num.quantile(percentiles, method=percentiles_method),
   1560                  num.max(split_every=split_every)]
   1561         stats_names = [(s._name, 0) for s in stats]

~/anaconda3/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in quantile(self, q, axis, method)
   1520         else:
   1521             _raise_if_object_series(self, "quantile")
-> 1522             meta = self._meta.quantile(q, axis=axis)
   1523             num = self._get_numeric_data()
   1524             quantiles = tuple(quantile(self[c], q, method) for c in num.columns)

TypeError: quantile() got an unexpected keyword argument 'axis'
```

**Expected behavior**
We should just take `axis` as default parameter which will be non-functional for our API to work with dask cudf.

**Comment by kkraus14** [comment]
@galipremsagar do you want to tackle this?

**Comment by galipremsagar** [comment]
@kkraus14 , Yepp I'll take it up. In the first place after doing a parameter fix there seems to be an exception coming up at another level of call stack. In the first glance to me it looked like a shape or dimension issue. But I'll take this up and fix it.

**Comment by galipremsagar** [comment]
Fixed by #2086
